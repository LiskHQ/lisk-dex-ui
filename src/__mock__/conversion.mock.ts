export const mockConversionRates = {
  LSK: {
    USD: 0.8863,
    DEX: 0.1,
  },
  DEX: {
    USD: 0.0886
  }
};