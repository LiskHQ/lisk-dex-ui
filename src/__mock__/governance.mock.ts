import { ProposalType } from "consts";
import { IPoolItem, IProposal } from "models";

export const mockProposal: IProposal = {
  author: "Author",
  proposalType: ProposalType.PoolIncentivization,
  poolID: "1",
  multiplier: 8,
  title: "title",
  summary: "summary",
  description: "description",
  link: "description"
}

export const mockPoolItems: IPoolItem[] = [
  {
    id: '1',
    chain1: 'LSK',
    chain2: 'DEU',
    rate: 0.3,
    amount: 12.2,
  },
  {
    id: '2',
    chain1: 'KLP',
    chain2: 'RGB',
    rate: 0.1,
    amount: 11.4,
  },
  {
    id: '3',
    chain1: 'DEU',
    chain2: 'FAE',
    rate: 0.3,
    amount: 17.6,
  },
  {
    id: '4',
    chain1: 'RGB',
    chain2: 'FAE',
    rate: 0.1,
    amount: 11.4,
  }
]