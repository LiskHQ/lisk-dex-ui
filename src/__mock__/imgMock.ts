export const svgUrl: string = 'svg-url';