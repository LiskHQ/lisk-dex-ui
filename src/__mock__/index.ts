export * from './governance.mock';
export * from './user.mock';