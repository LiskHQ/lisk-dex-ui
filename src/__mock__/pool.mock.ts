import { IPool } from "models";
import { mockTokens } from "./swap.mock";

export const mockPool: IPool = {
  token1: mockTokens[0],
  token2: mockTokens[1],
  token1Amount: 1862,
  token2Amount: 0.045,
  share: 0.085,
}

export const mockPools: IPool[] = [
  mockPool,
]