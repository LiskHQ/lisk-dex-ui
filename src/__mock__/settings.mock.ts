import { flagSvgs } from 'imgs/icons';
import { IFiatCurrency } from 'models';

export const mockFiatCurrnecies: IFiatCurrency[] = [
  {
    symbol: 'USD',
    name: 'US Dollar',
    image: flagSvgs.US,
  },
  {
    symbol: 'EUR',
    name: 'EU Euro',
    image: flagSvgs.EU,
  }
];