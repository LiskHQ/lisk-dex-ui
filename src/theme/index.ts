export { default as theme } from "./theme";
export { default as variables } from "./variables";
