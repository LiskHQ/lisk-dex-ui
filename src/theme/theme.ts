import {
  createTheme,
  responsiveFontSizes,
  ThemeOptions,
} from "@mui/material/styles";

import variables from "./variables";

const options: ThemeOptions = {
  palette: {
    primary: variables.primary,
    secondary: variables.secondary,
  },
  typography: {
    h1: {
      fontSize: "2.5rem",
      fontWeight: 600,
    },
    h2: {
      fontSize: "2rem",
      fontWeight: 400,
    },
    h3: {
      fontSize: "1.75rem",
      fontWeight: 400,
    },
    h4: {
      fontSize: "1.25rem",
      fontWeight: 400,
    },
    h5: {
      fontSize: "1rem",
      fontWeight: 400,
    },
    h6: {
      fontSize: "0.75rem",
      fontWeight: 400,
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 576,
      md: 768,
      lg: 1024,
      xl: 1440,
      xxl: 1950,
      xxxl: 2300,
    },
  },
};

const theme = responsiveFontSizes(createTheme(options));
export default theme;
