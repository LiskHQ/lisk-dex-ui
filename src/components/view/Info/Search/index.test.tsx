import React from 'react';
import { ThemeProvider } from '@mui/material';
import { fireEvent, render } from '@testing-library/react';
import { ISearchComponentProps, SearchComponent } from './index';
import { lightTheme } from 'styles/theme';
import { mockPoolDetails, mockTokenDetails } from '__mock__';

function renderComponent(props: ISearchComponentProps) {
  return render(
    <ThemeProvider theme={lightTheme}>
      <SearchComponent {...props} />
    </ThemeProvider>
  );
}

describe('SearchComponent page', () => {
  const mockProps: ISearchComponentProps = {
    onChangeSearchFilter: jest.fn(),
  };

  it('check if the component matches the snapshot', () => {
    const { container } = renderComponent(mockProps);
    expect(container).toMatchSnapshot();
  });

  it('check search input works fine', () => {
    const { getByTestId } = renderComponent(mockProps);
    fireEvent.input(getByTestId('search-input-test'), 'LSK');
    expect(mockProps.onChangeSearchFilter).toBeCalled();
  });

  it('check view pools more works fine', () => {
    const { getByTestId, getByPlaceholderText } = renderComponent({
      pools: mockPoolDetails,
      ...mockProps
    });
    const input = getByPlaceholderText('Search tokens or pools...');
    fireEvent.change(input, { target: { value: 'LSK' } });
    fireEvent.click(getByTestId('pools-view-more-test'));
    expect(input).toHaveValue('');
  });

  it('check view tokens more works fine', () => {
    const { getByTestId, getByPlaceholderText } = renderComponent({
      tokens: mockTokenDetails,
      ...mockProps
    });
    const input = getByPlaceholderText('Search tokens or pools...');
    fireEvent.change(input, { target: { value: 'LSK' } });
    fireEvent.click(getByTestId('tokens-view-more-test'));
    expect(input).toHaveValue('');
  });
});
