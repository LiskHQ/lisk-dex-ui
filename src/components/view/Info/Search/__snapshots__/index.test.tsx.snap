// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent page check if the component matches the snapshot 1`] = `
.emotion-0.filtered {
  top: -1.5rem;
  right: -1.5rem;
  position: absolute;
  padding: 1.5rem;
  border-radius: 0.25rem;
  background: #FFFFFF;
  z-index: 20;
}

.emotion-0 .search-input {
  width: 20.25rem;
  background: #F2F5F9;
}

.emotion-0 .search-input .MuiInputBase-root {
  border: none;
}

.emotion-0 .search-input input {
  margin-left: 0.25rem;
}

.emotion-0 .search-input .cancel-icon {
  cursor: pointer;
}

.emotion-0 .search-label {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-top: 1.5rem;
  margin-bottom: 1rem;
}

.emotion-0 .search-label .label-tvl {
  color: #6B7280;
}

.emotion-0 .search-pool-item {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0.25rem;
}

.emotion-0 .search-pool-item .search-pool-name {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 .search-pool-item .search-pool-name .pool-images {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 .search-pool-item .search-pool-name .pool-images .pool-image-2 {
  margin-left: -0.75rem;
}

.emotion-0 .search-pool-item .search-pool-name .pool-token-names {
  margin-left: 0.75rem;
}

.emotion-0 .search-pool-item .search-pool-name .pool-token-names p {
  font-size: 0.75rem;
  color: #6B7280;
}

.emotion-0 .search-pool-item .search-pool-value p {
  font-size: 0.75rem;
  color: #FF4557;
  text-align: right;
}

.emotion-0 .search-pool-item .search-pool-value p.increasement {
  color: #459A33;
}

.emotion-0 .search-pool-item .search-pool-value svg {
  margin-left: 0.25rem;
}

.emotion-0 .view-more {
  margin-top: 0.75rem;
  color: #6953F4;
  cursor: pointer;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  position: relative;
}

.emotion-1 .MuiInputLabel-root {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: unset;
  color: #282828!important;
  font-size: 0.875rem;
  line-height: 1.124rem;
  font-weight: 400;
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
  -webkit-transform: none;
  -ms-transform: none;
  margin-bottom: 0.5rem;
}

.emotion-1 .MuiInputLabel-root svg {
  cursor: pointer;
  margin-left: 0.25rem;
}

.emotion-1 .MuiInputLabel-root svg path {
  stroke: #6B7280!important;
}

.emotion-1 .MuiInputBase-root {
  padding: 0.75rem 1rem;
  border-radius: 0.25rem;
  color: #282828;
  border: 0.5px solid #E6E6E6;
  background: unset;
  -webkit-animation-duration: 1s!important;
  animation-duration: 1s!important;
  font-size: 1rem;
  line-height: 1.625rem;
}

.emotion-1 .MuiInputBase-root input {
  padding: 0;
}

.emotion-1 .MuiInputBase-root fieldset {
  border: none;
}

.emotion-1 .MuiInputBase-root:before {
  content: none;
}

.emotion-1 .MuiInputBase-root:after {
  content: none;
}

.emotion-1 .MuiInputBase-root:hover {
  border: 1px solid #A196F6;
}

.emotion-1 .MuiInputBase-root:focus-within {
  border: 1px solid #6953F4;
}

.emotion-1 .MuiFormHelperText-root {
  color: #FF4557;
  margin: 0.5rem 0 0 0;
}

.emotion-1 .MuiFormHelperText-root p {
  text-align: right;
}

.emotion-1 .input-length-counter {
  position: absolute;
  bottom: -1.5rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: end;
  justify-content: end;
  color: #6B7280;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.emotion-1 input::-webkit-outer-spin-button,
.emotion-1 input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  display: none;
}

.emotion-2 {
  font-size: 1rem;
  line-height: 1.4375em;
  font-weight: 500;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-3 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
}

.emotion-3::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3:focus {
  outline: 0;
}

.emotion-3:invalid {
  box-shadow: none;
}

.emotion-3::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-3.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-3:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-3::-webkit-input-placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-3::-moz-placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-3:-ms-input-placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-3::placeholder {
  opacity: 1;
  color: #6B7280;
}

<div>
  <div
    class="undefined  emotion-0"
  >
    <div
      class="MuiFormControl-root search-input emotion-1"
    >
      <div
        class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart emotion-2"
        data-testid="search-input-test"
      >
        <svg
          fill="none"
          height="20"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_1371_5467)"
          >
            <path
              d="M9.16667 15.8333C12.8486 15.8333 15.8333 12.8486 15.8333 9.16667C15.8333 5.48477 12.8486 2.5 9.16667 2.5C5.48477 2.5 2.5 5.48477 2.5 9.16667C2.5 12.8486 5.48477 15.8333 9.16667 15.8333Z"
              stroke="#9A9A9A"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.66667"
            />
            <path
              d="M17.5 17.5L13.875 13.875"
              stroke="#9A9A9A"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.66667"
            />
          </g>
          <defs>
            <clippath
              id="clip0_1371_5467"
            >
              <rect
                fill="white"
                height="20"
                width="20"
              />
            </clippath>
          </defs>
        </svg>
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiInputBase-inputAdornedStart emotion-3"
          placeholder="Search tokens or pools..."
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
